src/server.c

```diff
@@ -726,16 +728,75 @@ int dqlite_node_stop(dqlite_node *d)
 int dqlite_node_recover(dqlite_node *n,
 			struct dqlite_node_info infos[],
 			int n_info)
+{
+    int i;
+    int ret;
+
+    struct dqlite_node_info_ext *infos_ext = calloc((size_t)n_info, sizeof(*infos_ext));
+    if (infos_ext == NULL) {
+        return DQLITE_NOMEM;
+    }
+    for (i = 0; i < n_info; i++) {
+        infos_ext[i].size = sizeof(*infos_ext);
+        infos_ext[i].id = infos[i].id;
+        infos_ext[i].address = (uint64_t)(uintptr_t)infos[i].address;
```

Comment:
In LXC we have macros for this like `#define PTR_TO_UINT64(p) ((uint64_t)((uintptr_t)(p)))`. Might come in handy here.

Reply:
Thanks, that should indeed be more readable!